third_party/move/move-compiler-v2/src/insensitive_checker.rs

```diff
@@ -0,0 +1,158 @@
+        use ExpData::*;
+        match e {
+            Lambda(_, pat, _) | Block(_, pat, _, _) => {
+                if !post {
+                    self.enter_scope();
+                } else {
+                    // postorder
+                    for (id, var) in pat.vars() {
+                        self.node_symbol_decl_visitor(post, &id, &var);
+                    }
+                    self.exit_scope();
+                }
+            },
+            Quant(_, _, ranges, ..) => {
+                if !post {
+                    self.enter_scope();
+                } else {
+                    // postorder
+                    for (id, var) in ranges.iter().flat_map(|(pat, _)| pat.vars().into_iter()) {
+                        self.node_symbol_decl_visitor(post, &id, &var);
+                    }
+                    self.exit_scope();
+                }
+            },
+            Assign(_, pat, _) => {
+                for (id, sym) in pat.vars().iter() {
+                    self.node_symbol_use_visitor(post, id, sym);
+                }
+            },
+            LocalVar(id, sym) => {
+                self.node_symbol_use_visitor(post, id, sym);
+            },
+            Temporary(id, idx) => {
+                self.node_tmp_use_visitor(post, id, idx);
+            },
+            _ => {},
+        }
+        true // always continue
+    }
+
+    fn check_parameter_usage(&mut self) {
+        let symbol_pool = self.env.symbol_pool();
+        for (idx, Parameter(ref sym, _atype, ref loc)) in self.params.iter().enumerate() {
+            if !symbol_pool.symbol_starts_with_underscore(*sym)
+                && !self.used_tmps.contains(&idx)
+                && !self.seen_uses.contains(sym)
+            {
+                let msg = format!("Unused parameter '{}'.  Consider removing or prefixing with an underscore: '_{}'",
+                                  sym.display(symbol_pool),
+                                  sym.display(symbol_pool));
+                self.env.diag(Severity::Warning, loc, &msg);
+            }
+        }
+    }
+
+    fn enter_scope(&mut self) {
+        self.seen_uses_stack
+            .push(std::mem::take(&mut self.seen_uses));
+        self.seen_uses = BTreeSet::new();
+    }
+
+    fn exit_scope(&mut self) {
+        let mut new_seen_uses = self
+            .seen_uses_stack
+            .pop()
+            .expect("exits should balance enters");
+        self.seen_uses.append(&mut new_seen_uses);
+    }
+
+    fn node_symbol_decl_visitor(&mut self, post: bool, id: &NodeId, sym: &Symbol) {
+        if post {
+            let symbol_pool = self.env.symbol_pool();
+            if !self.seen_uses.contains(sym) {
+                if !symbol_pool.symbol_starts_with_underscore(*sym) {
+                    let loc = self.env.get_node_loc(*id);
+                    let msg = format!("Unused local variable '{}'.  Consider removing or prefixing with an underscore: '_{}'",
```

Comment:
nits: (1) use ` instead of ' when quoting variable names, (2) use single space after period. This applies to the other warning message above as well.

Reply:
Oh yeah.  (1) I was copying V1's messages.  Will fix it. (2) https://www.theatlantic.com/science/archive/2018/05/two-spaces-after-a-period/559304/.